src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -338,8 +338,12 @@ public boolean authenticate(final RestRequest request, final RestChannel channel
             }
 
             if(authCredenetials == null && anonymousAuthEnabled) {
-            	threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, User.ANONYMOUS);
-            	auditLog.logSucceededLogin(User.ANONYMOUS.getName(), false, null, request);
+                final String tenant = Utils.coalesce(request.header("securitytenant"), request.header("security_tenant"));
```

Comment:
Thank you for adding a test! The test looks good to me. One last question, what happens if the request does not contain the securitytenant header? Should it skip setting the requested tenant or is it ok that the value is set with empty/null?

Reply:
The behavior is the same of any other user. If you check the code for authenticated user, this header will be null if it's not set for that user. I don't mind to change it, I was just trying to be consistent with the same pattern used for the authenticated user.